src/WrappedMToken.sol

```diff
@@ -588,14 +628,17 @@ contract WrappedMToken is IWrappedMToken, Migratable, ERC20Extended {
      * @param currentIndex_ The current index.
      */
     function _startEarningFor(address account_, uint128 currentIndex_) internal {
-        _revertIfNotApprovedEarner(account_);
+        (bool isEarner_, uint16 feeRate_, ) = IEarnerManager(earnerManager).getEarnerDetails(account_);
+
+        if (!isEarner_) revert NotApprovedEarner(account_);
 
         Account storage accountInfo_ = _accounts[account_];
 
         if (accountInfo_.isEarning) return;
```

Comment:
if an account is already earning, and the integrator decides to adjust feeRate from 0 to %, it won't work